zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/AuthorizationCheckBehavior.java

```diff
@@ -183,113 +191,77 @@ private Optional<String> getUsername(final TypedRecord<?> command) {
-                membershipState.hasRelation(
-                    EntityType.ROLE, roleId, RelationType.TENANT, tenantId));
-  }
-
-  private Set<String> getTenantIdsForGroups(final Collection<String> groupIds) {
-    final var tenantIds = new HashSet<String>();
-    for (final var groupId : groupIds) {
-      tenantIds.addAll(
-          membershipState.getMemberships(EntityType.GROUP, groupId, RelationType.TENANT));
-
-      final var roleIds =
-          new HashSet<>(
-              membershipState.getMemberships(EntityType.GROUP, groupId, RelationType.ROLE));
-      tenantIds.addAll(getTenantIdsForRoles(roleIds));
-    }
-    return tenantIds;
-  }
-
-  private Set<String> getTenantIdsForRoles(final Collection<String> roleIds) {
-    final var tenantIds = new HashSet<String>();
-    for (final var roleId : roleIds) {
-      tenantIds.addAll(
-          membershipState.getMemberships(EntityType.ROLE, roleId, RelationType.TENANT));
-    }
-    return tenantIds;
+                        Stream.concat(
+                            membershipState
+                                .getMemberships(EntityType.GROUP, groupId, RelationType.TENANT)
+                                .stream(),
+                            membershipState
+                                .getMemberships(EntityType.GROUP, groupId, RelationType.ROLE)
+                                .stream()
+                                .flatMap(
+                                    roleId ->
+                                        membershipState
+                                            .getMemberships(
+                                                EntityType.ROLE, roleId, RelationType.TENANT)
+                                            .stream()))),
+            membershipState.getMemberships(entityType, entityId, RelationType.ROLE).stream()
+                .flatMap(
+                    roleId ->
+                        membershipState
+                            .getMemberships(EntityType.ROLE, roleId, RelationType.TENANT)
+                            .stream())));
   }
 
   public Set<String> getAllAuthorizedResourceIdentifiers(final AuthorizationRequest request) {
-    if (!authorizationsEnabled) {
-      return Set.of(WILDCARD_PERMISSION);
-    }
-
-    if (isAuthorizedAnonymousUser(request.getCommand())) {
+    if (!authorizationsEnabled || isAuthorizedAnonymousUser(request.getCommand())) {
       return Set.of(WILDCARD_PERMISSION);
     }
 
     return getUsername(request)
         .map(
             username ->
-                getUserAuthorizedResourceIdentifiers(
-                    username, request.getResourceType(), request.getPermissionType()))
-        .orElseGet(() -> getMappingsAuthorizedResourceIdentifiers(request))
+                getAuthorizedResourceIdentifiers(
+                    EntityType.USER,
+                    username,
+                    request.getResourceType(),
+                    request.getPermissionType()))
+        .or(
```

Comment:
❓ Assuming I'm reading this code correctly, we would want this combined and not individual branches, when using OIDC I could have both simple mappings (owned by username or application id) but also have advanced mappings (owned by mappings)

Reply:
Good catch :+1: That was the previous behavior too, I didn't change that. But a fix makes sense, I'll add it to this PR.

Reply:
Done with 5b929e4